src/components/WeatherApp.jsx

```diff
@@ -1,111 +1,205 @@
+          position.coords.latitude,
+          position.coords.longitude
+        ),
+      (error) => console.log(error)
+    );
+  };
+
+  return (
+    <div>
+      <main>
+        <div className="weatherApp">
+          <div className="searchSection">
+            <input
+              type="text"
+              className="inputText"
+              placeholder="Search..."
+              value={inputName}
+              onChange={(e) => setInputName(e.target.value)}
+              onKeyDown={handleInputBoxKeydown}
+            />
+            <div className="location-icon">
+              <FontAwesomeIcon icon={faLocationDot} onClick={handleLocation} />
+            </div>
+            <div className="searchIcon">
+              <FontAwesomeIcon
+                icon={faMagnifyingGlass}
+                onClick={onClickRenderWeather}
+              />
+            </div>
+          </div>
+          <div>
+            <img
+              src="https://previews.123rf.com/images/martialred/martialred1702/martialred170200035/71710170-cloudy-or-cloud-partly-blocking-the-sun-flat-color-vector-icon-for-weather-apps-and-websites.jpg"
+              alt="weather Img"
+              className=" weatherImg"
+            />
+          </div>
+          <div>
+            <div className="subMainSection">
+              <div className="tempMainDiv ">
+                <div className="tempSubDiv">
+                  <h2
+                    className={
+                      weatherData.error
+                        ? "h2-displayErrorSymbol"
+                        : "h2-removeErrorSymbol"
+                    }
+                  >
+                    {isLoading ? (
+                      <BeatLoader color="#252827" size={10} />
+                    ) : weatherData.error ? (
+                      "?"
+                    ) : (
+                      weatherData?.current?.temp_c
+                    )}{" "}
+                  </h2>
+                  <span>&#8451;</span>
                 </div>
-            </main>
-            <ToastContainer />
+                <p
+                  className={
+                    weatherData.error
+                      ? "tempSub-p-displayErrorSymbol"
+                      : "tempSub-p-removeErrorSymbol"
+                  }
+                >
+                  {isLoading ? (
+                    <BeatLoader color="#252827" size={10} />
+                  ) : weatherData.error ? (
+                    "?"
+                  ) : (
+                    weatherData?.current?.condition?.text
+                  )}
+                </p>
+              </div>
+            </div>
+            <div className="lastSection">
+              <div className="weatherSection">
+                <div>
+                  {isLoading ? (
+                    <BeatLoader color="#252827" size={10} />
+                  ) : weatherData.error ? (
+                    "?"
+                  ) : weatherData?.current?.humidity <= 55 ? (
```

Comment:
Lets make a separate component for this. Should look like this: 
<RainDropRenderer humidity={30} />
Inside that component , take the humidity and render the relevant drop icon